variation-commons-core/src/test/java/uk/ac/ebi/eva/commons/core/models/factories/VariantVcfFactoryTest.java

```diff
@@ -207,31 +216,42 @@ public void testCreateVariantFromVcfCoLocatedVariants_MainFields() {
 
     @Test
     public void testVariantIds() {
-
-        Set<String> emptySet = new HashSet<>();
-        // test that an ID is properly handled
-        String line = "1\t1000\trs123\tC\tT\t.\t.\t.";
-        List<Variant> expResult = new LinkedList<>();
-        expResult.add(new Variant("1", 1000, 1000, "C", "T"));
-        List<Variant> result = factory.create(FILE_ID, STUDY_ID, line);
-        assertEquals(expResult, result);
         //EVA-942 - Since we ignore IDs submitted through VCF, they are no longer part of the expected result
-        assertEquals(emptySet, result.get(0).getIds());
+        Set<String> emptySet = new HashSet<>();
 
-        // test that the ';' is used as the ID separator (as of VCF 4.2)
-        line = "1\t1000\trs123;rs456\tC\tT\t.\t.\t.";
-        expResult = new LinkedList<>();
-        expResult.add(new Variant("1", 1000, 1000, "C", "T"));
-        result = factory.create(FILE_ID, STUDY_ID, line);
-        assertEquals(expResult, result);
-        assertEquals(emptySet, result.get(0).getIds());
+        // test that an ID is ignored
+        checkIds(factory, "1\t1000\trs123\tC\tT\t.\t.\t.", emptySet);
+
+        // test that several ID are ignored
+        checkIds(factory, "1\t1000\trs123;rs456\tC\tT\t.\t.\t.", emptySet);
 
         // test that a missing ID ('.') is not added to the IDs set
-        line = "1\t1000\t.\tC\tT\t.\t.\t.";
-        expResult = new LinkedList<>();
-        expResult.add(new Variant("1", 1000, 1000, "C", "T"));
-        result = factory.create(FILE_ID, STUDY_ID, line);
-        assertEquals(expResult, result);
-        assertEquals(emptySet, result.get(0).getIds());
+        checkIds(factory, "1\t1000\t.\tC\tT\t.\t.\t.", emptySet);
+
+
+        // needed for eva-accession-clustering, test that ID is read if explicitly configured
+        VariantVcfFactory accessionedVariantFactory = instantiateAbstractVcfFactory();
+        accessionedVariantFactory.setIncludeIds(true);
+
+        // test that an ID is properly read
+        checkIds(accessionedVariantFactory, "1\t1000\trs123\tC\tT\t.\t.\t.", Collections.singleton("rs123"));
+
+        // test that a missing ID ('.') is not added to the IDs set
+        checkIds(accessionedVariantFactory, "1\t1000\trs123;.\tC\tT\t.\t.\t.", Collections.singleton("rs123"));
+
+        // test that the ';' is used as the ID separator (as of VCF 4.2)
+        checkIds(accessionedVariantFactory, "1\t1000\trs123;rs456\tC\tT\t.\t.\t.",
```

Comment:
Another check to ensure that RS IDs separated by comma are read as-is. While this is not ideal, at least we make it explicit that it is the current behavior...